libraries/chain/include/eosio/chain/wasm_interface.hpp

```diff
@@ -62,6 +65,17 @@ namespace eosio { namespace chain {
          //Immediately exits currently running wasm. UB is called when no wasm running
          void exit();
 
+         // If substitute_apply is set, then apply calls it before doing anything else. If substitute_apply returns true,
+         // then apply returns immediately.
+         std::function<bool(
+            const digest_type& code_hash, uint8_t vm_type, uint8_t vm_version, apply_context& context)> substitute_apply;
+
+         // If substitute_module is set, then get_instantiated_module calls it to instantiate a module. If it returns a
+         // module, then get_instantiated_module caches the result. If it returns nullptr, get_instantiated_module
+         // operates normally.
+         std::function<std::unique_ptr<wasm_instantiated_module_interface>(
+            const digest_type& code_hash, uint8_t vm_type, uint8_t vm_version)> substitute_module;
```

Comment:
I don't quite see why both of these are needed.  Isn't either one sufficient to hijack the execution?

Reply:
Yes, especially now that cltester handles wasm caching to prevent it having to spend time regenerating the debug info with each new chain instance.

Reply:
I shouldn't have gone so long without testing with gcc.

* gcc 9, native debug build: everything seems to work
* gcc 9, native release build: `*** longjmp causes uninitialized stack frame ***: terminated`

I'll upgrade to 10 to see if the longjmp issue hangs around

Reply:
Whoops. Wrong window.